### PR TITLE
Reduce TSV Sizes

### PR DESCRIPTION
**Pull request by afontani**
## Pull Request Description

**Companion PR:** resstock-estimation [#200](https://github.com/NREL/resstock-estimation/pull/200)

As housing characteristics get bigger, we are moving to a compact writing style. The 6-digit float format requirement was put in place when most of the characteristics were not scripted. Now that there is a standard workflow for creating the characteristics, the formatting requirements are being relaxed. This change should allow for more accurate characteristic distributions because the exponential format can be used. The change should also stop round-off errors.

The change is mainly from `%.6f` to `%g` in `pandas.to_csv(...)` in the resstock-estimation repository.

Some CI tests were removed in the integrity checks that related to formatting checks in the characteristics.

## Housing Characteristic size reduction

**Project National**
- Before Change: 103MB
- After Change: 46MB

**Project Testing**
- Before Change: 648KB
- After Change: 560KB

## Checklist

Not all may apply:

- [x] Tests (and test files) have been updated
- [x] Documentation has been updated
- [x] Changelog has been updated
~~[ ] `openstudio tasks.rb update_measures` has been run~~
- [ ] No unexpected integration/regression test changes

### CODE REVIEW

File: test/integrity_checks.rb [hunk 1]
```diff
@@ -333,31 +333,7 @@ def check_parameter_file_format(tsvpath, n_deps, name)
...
-            raise "ERROR: Incorrect non float found in '#{name}', line '#{i}'."
-          end
-          # If float precision is not 6 digits, raise error
-          if float_precision != 6
-            raise "ERROR: Incorrect float precision found in '#{name}', line '#{i}'."
-          end
-        end
-      end
+      next
     else
```
Comment (shorowit) [inline test/integrity_checks.rb:337]: There is a small bug here, where by using `next`, the `i += 1` below will not get executed.

Suggest changing this from:
```
    if line.include? "\r\n"
      next
    else
```
to
```
    if not line.include? "\r\n"
```

**Review by shorowit — APPROVED** [review]
Looks good to me. Nice job updating all related tests, documentation, etc. Just one minor bug to address.

File: test/integrity_checks.rb (hunk 1 above)
Comment (afontani) [reply test/integrity_checks.rb:337]: Thanks @shorowit!

File: test/integrity_checks.rb [hunk 2]
```diff
@@ -332,33 +332,7 @@ def check_parameter_file_format(tsvpath, n_deps, name)
...
-      if i > 1
-        # Check float format
-        # Remove endline character and split the string into array
-        line = line.split("\r\n")[0].split("\t")
-        # For each non dependency entry check format
-        for j in n_deps..line.length() - 1 do
-          # Check for scientific format
-          if (line[j].include?('e-') || line[j].include?('e+') ||
-              line[j].include?('E-') || line[j].include?('E+'))
-            raise "ERROR: Scientific notation found in '#{name}', line '#{i}'."
```
Comment (joseph-robertson) [inline test/integrity_checks.rb:346]: Should we remove this check?

**Comment by afontani** [comment]
> Do you have the numbers on how much this reduces total file size?

@joseph-robertson: Good suggestion. I updated the PR description. Project national reduced by about 50%.

File: test/integrity_checks.rb (hunk 2 above)
Comment (afontani) [reply test/integrity_checks.rb:346]: The `%g` format allows for scientific notation. I actually think this is a good thing as it handles smaller values really well and helps with truncation error.

File: test/integrity_checks.rb [hunk 3]
```diff
@@ -332,33 +332,7 @@ def check_parameter_file_format(tsvpath, n_deps, name)
...
-          end
-
-          begin # Try to get the float precision
-            float_precision = line[j].split('.')[1].length()
-          rescue NoMethodError
-            # Catch non floats
-            raise "ERROR: Incorrect non float found in '#{name}', line '#{i}'."
```
[the remark this answers is not in the document]
Comment (afontani) [reply test/integrity_checks.rb:353]: @joseph-robertson: I had to think about this for a second, but in the new format integers are acceptable. An example is our mapping characteristics (ex: State is dependant on County and results in a bunch of 1s and 0s). 

There is still a sum to 1.0 and positivity check, which should check for nans and odd values.

**Review by joseph-robertson — APPROVED** [review]

**Pull request merged**